grpc/optimistic/mock_grpc_stream.go

```diff
@@ -0,0 +1,91 @@
+package optimistic
+
+import (
+	"context"
+	"google.golang.org/grpc/metadata"
+	"io"
+	"time"
+)
+
+type MockBidirectionalStreaming[K any, V any] struct {
+	requestStream        []*K
+	accumulatedResponses []*V
+	requestCounter       uint64
+}
+
+func (ms *MockBidirectionalStreaming[K, V]) Recv() (*K, error) {
+	// add a delay to make it look like an async stream
+	time.Sleep(2 * time.Second)
+	if ms.requestCounter > uint64(len(ms.requestStream)-1) {
+		// end the stream after all the packets have been sent
+		return nil, io.EOF
+	}
+
+	req := ms.requestStream[ms.requestCounter]
+	ms.requestCounter += 1
+
+	return req, nil
+}
+
+func (ms *MockBidirectionalStreaming[K, V]) Send(res *V) error {
+	ms.accumulatedResponses = append(ms.accumulatedResponses, res)
+	return nil
+}
+
+func (ms *MockBidirectionalStreaming[K, V]) SetHeader(md metadata.MD) error {
+	panic("implement me")
+}
+
+func (ms *MockBidirectionalStreaming[K, V]) SendHeader(md metadata.MD) error {
+	panic("implement me")
+}
+
+func (ms *MockBidirectionalStreaming[K, V]) SetTrailer(md metadata.MD) {
+	panic("implement me")
+}
+
+func (ms *MockBidirectionalStreaming[K, V]) Context() context.Context {
+	return context.Background()
+}
+
+func (ms *MockBidirectionalStreaming[K, V]) SendMsg(m any) error {
+	panic("implement me")
+}
+
+func (ms *MockBidirectionalStreaming[K, V]) RecvMsg(m any) error {
+	panic("implement me")
+}
+
+type MockServerSideStreaming[K any] struct {
+	sentResponses []*K
+}
+
+func (ms *MockServerSideStreaming[K]) SendMsg(m any) error {
+	//TODO implement me
```

Comment:
does this need to be done?

Reply:
no no, these methods are not being used in the streaming methods so we don't have to implement them! I ll write better comments on these!

Reply:
addressed in https://github.com/astriaorg/flame/pull/30/commits/defc35bb0c32331a823a635dccdb28c7e62d519b